### PR TITLE
refactor: fix goerr113 linter

### LINKED ISSUE #541: lint: fix goerr113
* Fix `goerr113` warnings from the [lint checker](https://github.com/ovrclk/akash/blob/master/.github/workflows/tests.yml#L39) step for it.
 * Remove `continue-on-error: true` from `lint goerr113` step.

### PR DESCRIPTION
**Pull request by troian**
fixes #541

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -8,5 +8,8 @@ run:
   skip-dirs:
     - pkg/client/clientset
 
+  skip-files:
+    - pkg/client/informers/externalversions/generic.go
```
Comment (boz) [inline .golangci.yaml:12]: I think that everything in and below `pkg/client` is auto-generated.

File: cmd/akashd/genaccounts.go [hunk 2]
```diff
@@ -98,26 +102,26 @@ contain valid denominations. Accounts may optionally be supplied with vesting pa
...
-					return errors.New("invalid vesting parameters; must supply start and end time or end time")
+					return errors.Wrap(ErrInvalidVestingParameters, "must supply start and end time or end time")
 				}
 			} else {
 				genAccount = baseAccount
 			}
 
-			if err := genAccount.Validate(); err != nil {
-				return fmt.Errorf("failed to validate new genesis account: %w", err)
+			if err = genAccount.Validate(); err != nil {
```
Comment (boz) [inline cmd/akashd/genaccounts.go:111]: unnecessary change.  it should define a new error for that scope.

File: provider/cluster/kube/client.go [hunk 3]
```diff
@@ -28,6 +28,12 @@ import (
 	mtypes "github.com/ovrclk/akash/x/market/types"
 )
 
+var (
+	ErrNoDeploymentForLease = errors.New("kube: no deployments for lease")
+	ErrNoIngressForLease    = errors.New("kube: no ingress for lease")
+	ErrInternalError = errors.New("kube: internal error")
```
Comment (boz) [inline provider/cluster/kube/client.go:34]: formatting looks off - is your editor not aligning `=`?  we should have a linter for this.

File: provider/cluster/kube/client.go [hunk 4]
```diff
@@ -49,42 +55,39 @@ func NewClient(log log.Logger, host, ns string) (Client, error) {
...
 	metc, err := metricsclient.NewForConfig(config)
 	if err != nil {
-		return nil, fmt.Errorf("error creating metrics client: %v", err)
+		return nil, errors.Wrap(err, "kube: error creating metrics client")
 	}
 
-	err = akashv1.CreateCRD(ctx, mcr)
-	if err != nil {
-		return nil, fmt.Errorf("error creating akashv1 CRD: %v", err)
+	if err = akashv1.CreateCRD(ctx, mcr); err != nil {
```
Comment (boz) [inline provider/cluster/kube/client.go:81]: `:=`

File: provider/cluster/kube/client.go [hunk 5]
```diff
@@ -49,42 +55,39 @@ func NewClient(log log.Logger, host, ns string) (Client, error) {
...
+		return nil, errors.Wrap(err, "kube: error creating akashv1 CRD")
 	}
 
-	err = prepareEnvironment(ctx, kc, ns)
-	if err != nil {
-		return nil, fmt.Errorf("error preparing environment %v", err)
+	if err = prepareEnvironment(ctx, kc, ns); err != nil {
```
Comment (boz) [inline provider/cluster/kube/client.go:85]: `:=`

File: provider/cluster/kube/client.go [hunk 6]
```diff
@@ -49,42 +55,39 @@ func NewClient(log log.Logger, host, ns string) (Client, error) {
...
+		return nil, errors.Wrap(err, "kube: error preparing environment")
 	}
 
-	_, err = kc.CoreV1().Namespaces().List(ctx, metav1.ListOptions{Limit: 1})
-	if err != nil {
-		return nil, fmt.Errorf("error connecting to kubernetes: %v", err)
+	if _, err = kc.CoreV1().Namespaces().List(ctx, metav1.ListOptions{Limit: 1}); err != nil {
```
Comment (boz) [inline provider/cluster/kube/client.go:89]: `:=`

File: provider/cluster/kube/client.go [hunk 7]
```diff
@@ -188,18 +191,19 @@ func (c *client) ServiceLogs(ctx context.Context, lid mtypes.LeaseID,
...
 	if err != nil {
 		c.log.Error(err.Error())
-		return nil, errors.New("internal error")
+		return nil, errors.Wrap(err, ErrInternalError.Error())
 	}
 	streams := make([]*cluster.ServiceLog, len(pods.Items))
 	for i, pod := range pods.Items {
-		stream, err := c.kc.CoreV1().Pods(lidNS(lid)).GetLogs(pod.Name, &corev1.PodLogOptions{
+		var stream io.ReadCloser
+		stream, err = c.kc.CoreV1().Pods(lidNS(lid)).GetLogs(pod.Name, &corev1.PodLogOptions{
```
Comment (boz) [inline provider/cluster/kube/client.go:199]: why?

File: x/deployment/genesis.go [hunk 8]
```diff
@@ -45,7 +45,7 @@ func DefaultGenesisState() GenesisState {
 // InitGenesis initiate genesis state and return updated validator details
 func InitGenesis(ctx sdk.Context, keeper keeper.Keeper, data GenesisState) []abci.ValidatorUpdate {
 	for _, record := range data.Deployments {
-		keeper.Create(ctx, record.Deployment, record.Groups)
+		_ = keeper.Create(ctx, record.Deployment, record.Groups)
```
Comment (boz) [inline x/deployment/genesis.go:48]: We should probably `panic` here.

File: x/market/query/path.go [hunk 9]
```diff
@@ -18,6 +19,11 @@ const (
 	leasePath  = "lease"
 )
 
+var (
+	ErrInvalidPath = errors.New("query: invalid path")
+	ErrStateValue = errors.New("query: invalid state value")
```
Comment (boz) [inline x/market/query/path.go:24]: not formatted again.

File: provider/cluster/kube/client.go (hunk 7 above)
Comment (troian) [reply provider/cluster/kube/client.go:199]: nevermind, that was coming from errcheck dependency. rid off

File: validation/resources.go [hunk 10]
```diff
@@ -4,9 +4,16 @@ import (
...
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/pkg/errors"
+
 	"github.com/ovrclk/akash/types"
 )
 
+var (
+	ErrNoGroupsPresent = errors.New("validation: no groups present")
+	ErrGroupEmptyName = errors.New("validation: group has empty name")
```
Comment (boz) [inline validation/resources.go:14]: let's figure out why this isn't being caught by a linter @troian

Comment (troian) [reply validation/resources.go]: Fixed. the best option would be to run linter when making commit...

File: sdl/v1.go [hunk 11]
```diff
@@ -1,14 +1,14 @@
...
 import (
-	"fmt"
 	"sort"
 
 	"github.com/blang/semver"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/ovrclk/akash/manifest"
 	"github.com/ovrclk/akash/types"
 	dtypes "github.com/ovrclk/akash/x/deployment/types"
+	"github.com/pkg/errors"
```
Comment (Ropes) [inline sdl/v1.go:11]: nitpick(not necessary to fix): raise import line to be just below the standard library packages.

File: x/deployment/simulation/operations.go [hunk 12]
```diff
@@ -10,6 +9,8 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
 	"github.com/cosmos/cosmos-sdk/x/simulation"
+	"github.com/pkg/errors"
```
Comment (Ropes) [inline x/deployment/simulation/operations.go:12]: nit: swap it with the `"github.com/cosmos/cosmos-sdk/baseapp"` import line for descending import specificity.
